### PR TITLE
Add a walk state flag to skip methods with the JvmtiMountTransition annotation

### PR DESCRIPTION
**Pull request by thallium**
Since it's only related to JVMTI, a walk state flag is added to enable the behaviour when doing JVMTI-related stack walk.

Related: https://github.com/eclipse-openj9/openj9/issues/17490

### CODE REVIEW

**Comment by thallium** [comment]
@babsingh FYI

File: runtime/oti/j9.h [hunk 1]
```diff
@@ -374,6 +374,8 @@ static const struct { \
 
 #define J9VM_VIRTUALTHREAD_ROOT_NODE_STATE ((I_32)0xBAADF00D)
 
+#define J9_IS_JVMTI_MOUNT_TRANSITION(method) \
+	((NULL != (method)) && (J9_ARE_ANY_BITS_SET(getExtendedModifiersDataFromROMMethod(J9_ROM_METHOD_FROM_RAM_METHOD(method)), CFR_METHOD_EXT_JVMTIMOUNTTRANSITION_ANNOTATION)))
 #if defined(OPENJ9_BUILD)
```
Comment (babsingh) [inline runtime/oti/j9.h:379]: Add a new line.
```suggestion
#if JAVA_SPEC_VERSION >= 20
#define J9_IS_JVMTI_MOUNT_TRANSITION(method) \
	((NULL != (method)) && (J9_ARE_ANY_BITS_SET(getExtendedModifiersDataFromROMMethod(J9_ROM_METHOD_FROM_RAM_METHOD(method)), CFR_METHOD_EXT_JVMTIMOUNTTRANSITION_ANNOTATION)))
#endif /* JAVA_SPEC_VERSION >= 20 */

#if defined(OPENJ9_BUILD)
```

File: runtime/vm/swalk.c [hunk 2]
```diff
@@ -481,6 +481,13 @@ UDATA walkFrame(J9StackWalkState * walkState)
 			return J9_STACKWALK_KEEP_ITERATING;
 		}
 
+		/* Skip methods with JvmtiMountTransition annotation */
+		if (J9_ARE_ALL_BITS_SET(walkState->flags, J9_STACKWALK_SKIP_JVMTI_MOUNT_TRANSITION_FRAMES)
+			&& J9_IS_JVMTI_MOUNT_TRANSITION(walkState->method)
+		) {
+			return J9_STACKWALK_KEEP_ITERATING;
+		}
```
Comment (babsingh) [inline runtime/vm/swalk.c:489]: To be consistent with https://github.com/eclipse-openj9/openj9/pull/16654, the code changes in this PR should be wrapped with `JAVA_SPEC_VERSION >= 20`.
```suggestion
#if JAVA_SPEC_VERSION >= 20
		/* Skip methods with JvmtiMountTransition annotation */
		if (J9_ARE_ALL_BITS_SET(walkState->flags, J9_STACKWALK_SKIP_JVMTI_MOUNT_TRANSITION_FRAMES)
			&& J9_IS_JVMTI_MOUNT_TRANSITION(walkState->method)
		) {
			return J9_STACKWALK_KEEP_ITERATING;
		}
#endif /* JAVA_SPEC_VERSION >= 20 */
```

File: runtime/oti/j9consts.h [hunk 3]
```diff
@@ -401,6 +401,7 @@ extern "C" {
 #define J9_STACKWALK_CACHE_METHODS 0x400
 #define J9_STACKWALK_CACHE_MASK 0x700
 #define J9_STACKWALK_SKIP_HIDDEN_FRAMES 0x800
+#define J9_STACKWALK_SKIP_JVMTI_MOUNT_TRANSITION_FRAMES 0x1000
```
Comment (babsingh) [inline runtime/oti/j9consts.h:404]: Macro definitions don't need `JAVA_SPEC_VERSION >= 20`.

**Review by babsingh — CHANGES_REQUESTED** [review]

**Comment by gacholio** [comment]
This stuff really doesn't belong in the stack walker (I know previous hacks for hidden have made it in). Really, it should be handled in the frame iterators, much like we do to hide the old reflect implementation).

File: runtime/jvmti/jvmtiHelpers.c [hunk 4]
```diff
@@ -1813,7 +1813,7 @@ UDATA
 findDecompileInfo(J9VMThread *currentThread, J9VMThread *targetThread, UDATA depth, J9StackWalkState *walkState)
 {
 	walkState->walkThread = targetThread;
-	walkState->flags = J9_STACKWALK_ITERATE_FRAMES | J9_STACKWALK_INCLUDE_NATIVES | J9_STACKWALK_VISIBLE_ONLY | J9_STACKWALK_RECORD_BYTECODE_PC_OFFSET | J9_STACKWALK_MAINTAIN_REGISTER_MAP;
+	walkState->flags = J9_STACKWALK_ITERATE_FRAMES | J9_STACKWALK_INCLUDE_NATIVES | J9_STACKWALK_VISIBLE_ONLY | J9_STACKWALK_RECORD_BYTECODE_PC_OFFSET | J9_STACKWALK_MAINTAIN_REGISTER_MAP | J9_STACKWALK_SKIP_JVMTI_MOUNT_TRANSITION_FRAMES;
```
Comment (gacholio) [inline runtime/jvmti/jvmtiHelpers.c:1816]: This is certainly a place that should not be changed - this wants the true view of the stack, not a user-facing one.

**Comment by gacholio** [comment]
The only place using the new flag is using it incorrectly as far as I can see, so we should probably just abandon this PR.

**Pull request closed without merging**